src/Common/Requests/BaseRequest.php

```diff
@@ -0,0 +1,135 @@
+<?php
+/**
+ * Created by IntelliJ IDEA.
+ * User: emcnaughton
+ * Date: 5/4/17
+ * Time: 7:34 AM
+ */
+
+namespace Omnimail\Common\Requests;
+
+use Omnimail\Common\Helper;
+
+abstract class BaseRequest implements RequestInterface
+{
+
+    /**
+     * Timestamp for start of period.
+     *
+     * @var int
+     */
+    protected $startTimeStamp;
+
+    /**
+     * Timestamp for end of period.
+     *
+     * @var int
+     */
+    protected $endTimeStamp;
+
+    /**
+     * Url to direct requests to.
+     *
+     * @var string
+     */
+    protected $endPoint;
+
+    /**
+     * User name
+     *
+     * @var string
+     */
+    protected $username;
+
+    /**
+     * Password
+     *
+     * @var string
+     */
+    protected $password;
+
+    /**
+     * @return string
+     */
+    public function getEndPoint()
+    {
+        return $this->endPoint;
+    }
+
+    /**
+     * @param string $endPoint
+     */
+    public function setEndPoint($endPoint)
+    {
+        $this->endPoint = $endPoint;
+    }
+
+    /**
+     * @return int
+     */
+    public function getStartTimeStamp()
+    {
+        return $this->startTimeStamp;
+    }
+
+    /**
+     * @param int $startTimeStamp
+     */
+    public function setStartTimeStamp($startTimeStamp)
+    {
+        $this->startTimeStamp = $startTimeStamp;
+    }
+
+    /**
+     * @return int
+     */
+    public function getEndTimeStamp()
+    {
+        return $this->endTimeStamp;
+    }
+
+    /**
+     * @param int $endTimeStamp
+     */
+    public function setEndTimeStamp($endTimeStamp)
+    {
+        $this->endTimeStamp = $endTimeStamp;
+    }
+
+    /**
+     * @return string
+     */
+    public function getUsername()
```

Comment:
I think that makes sense - I have a feeling there should be a credentials object - but I'm struggling to articulate exactly why. Payum uses a Config object 

https://github.com/Payum/Payum/blob/master/src/Payum/Core/Model/GatewayConfig.php

and they provide encryption/ decryption functionality there. Even if we aren't adding that at this stage I feel like having credentials as an object leaves the door more open for that. They also do some stuff with their sensitiveInformationObject to help make data more secure

https://github.com/Payum/Payum/blob/master/src/Payum/Core/Security/SensitiveValue.php

It looks like the key parts are 
- using a private method 
- overriding serialize()
- deleting credit card data from the class once retrieved
- overriding __debugInfo() to prevent it being output in debugging.

The credentials are not as sensitive as credit card details - but these seem like nice techniques (that I was not aware of)